de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/api/impl/InvoiceCandBL.java

```diff
@@ -183,6 +183,7 @@
 	private static final LocalDate DATE_TO_INVOICE_MAX_DATE = LocalDate.of(9999, Month.DECEMBER, 31); // NOTE: not using LocalDate.MAX because we want to convert it to Timestamp
 
 	private final Logger logger = InvoiceCandidate_Constants.getLogger(InvoiceCandBL.class);
+	private final IUOMDAO uomsRepo = Services.get(IUOMDAO.class);
```

Comment:
AFIU adding a singletonservice like this as a bad practice because we have no mechanism (unlike spring) to update the implementation.
This has lead to hard-to-debug test failures in the past. I suggest you move 
```suggestion
```

Reply:
btw, wdyt about my proposal to distinguish between "repo" (sth that load an entity) and "DAO" (something that loads a data record)?